ttnn/cpp/ttnn/operations/core/to_layout/to_layout_op.cpp

```diff
@@ -42,6 +42,91 @@ inline bool use_multicore_device_tilize(
     return num_tiles_in_row <= max_tiles;
 }
 
+bool requires_padding_change(const ttnn::Tensor& tensor, ttnn::Layout layout) {
```

Comment:
maybe `requires_alignment_change`?

Reply:
I would keep it like this, it's easier to understand. Alignment change can mean a lot of things, changing padding means that I'm either adding or removing some rows/cols at the end.